.github/workflows/lint.yml

```
@@ -0,0 +1,59 @@
+---
+# Lint/analyse code with Super-Linter.
+name: Checks
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main, 'releases/**']
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress:  ${{ github.ref != 'refs/heads/main' }}
+
+permissions: read-all
+
+jobs:
+  check:
+    name: Lint
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 4
+      fail-fast: false
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
+
+    permissions:
+        contents: read
+        packages: read
+        statuses: write
+
+    steps:
+      - name: Checkout current
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Super-Linter
+        uses: super-linter/super-linter/slim@v7.2.1
```

Suggest we pin the version commit
```suggestion
        uses: super-linter/super-linter/slim@85f7611e0f7b53c8573cca84aa0ed4344f6f6a4d  # v7.2.1
```